Guide.docc/RuntimeBehavior.md

```diff
@@ -0,0 +1,207 @@
+task, before any other low-priority tasks. This effectively leads to FIFO order
+violations, because such task "jumped ahead" of other tasks which may have been
+enqueue on the actor well ahead of it. This does does help make actors very
+responsive to high priority work which is a valuable property to have!
+
+> Note: Priority escalation is not supported on actors with custom executors.
+
+The safest and correct way to enqueue a number of items to be processed by an actor,
+in a specific order is to use an `AsyncStream` to form a single, well-ordered
+sequence of items, which can be emitted to even from synchronous code.
+And then consume it using a _single_ task running on the actor, like this:
+
+```swift
+// ✅ Guaranteed order in which log() are invoked,
+//    regardless of priority escalations, because the disconnect 
+//    between the producing and consuming task
+actor Printer {
+    let stream: AsyncStream<Int>
+    let streamContinuation: AsyncStream<Int>.Continuation
+    var streamConsumer: Task<Void, Never>!
+
+    init() async {
+        let (stream, continuation) = AsyncStream.makeStream(of: Int.self)
+        self.stream = stream
+        self.streamContinuation = continuation
+
+        // Consuming Option A)
+        // Start consuming immediately, 
+        // or better have a caller of Printer call `startStreamConsumer()`
+        // which would make it run on the caller's task, allowing for better use of structured concurrency.
+        self.streamConsumer = Task { await self.consumeStream() }
+    }
+
+    deinit {
+        self.streamContinuation.finish()
+    }
+  
+    nonisolated func enqueue(_ item: Int) {
+        self.streamContinuation.yield(item)
+    }
+  
+    nonisolated func cancel() { 
+      self.streamConsumer?.cancel()
+    }
+
+    func consumeStream() async {
+        for await item in self.stream {
+            if Task.isCancelled { break }
+          
+            log(item)
+        }
+    }
+
+    func log(_ int: Int) { print(int) }
+}
+```
+
+and invoke it like:
+
+```
+let printer: Printer = ... 
+printer.enqueue(1)
+printer.enqueue(2)
+printer.enqueue(3)
+```
+
+We're assuming that the caller has to be in synchronous code, and this is why we make the `enqueue`
+method `nonisolated` but we use it to funnel work items into the actor's stream.
+
+The actor uses a single task to consume the async sequence of items, and this way guarantees
+the specific order of items being handled.
+
+This approach has both up and down-sides, because now the item processing cannot be affected by
+priority escalation. The items will always be processed in their strict enqueue order,
+and we cannot easily await for their results -- since the caller is in synchronous code,
+so we might need to resort to callbacks if we needed to report completion of an item
+getting processed.
+
+Notice that we kick off an unstructured task in the actor's initializer, to handle the
```

Comment:
Ah we mentioned this down here. Can we just not show the unstructured task to begin with?